### ISSUE TITLE
The compiler should complain about unknown features before complaining about missing crate attributes

### ISSUE THREAD

**Issue by r00ster91** [rust-lang/rust#83998]
Given the following code: 

```rust
#![feature(const_raw_ptr_dere)] // Note the missing f
#![feature(const_mut_refs)]

struct A;

const fn a() -> &'static mut A {
    unsafe { &mut *(0x11 as *mut A) }
}
```

The current output is:

```
error[E0658]: dereferencing raw pointers in constant functions is unstable
 --> src/lib.rs:7:14
  |
7 |     unsafe { &mut *(0x11 as *mut A) }
  |              ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #51911 <https://github.com/rust-lang/rust/issues/51911> for more information
  = help: add `#![feature(const_raw_ptr_deref)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: could not compile `playground`

To learn more, run the command again with --verbose.
```

Ideally the output should look like:

```
error[E0635]: unknown feature `const_raw_ptr_dere`
 --> src/lib.rs:1:12
  |
1 | #![feature(const_raw_ptr_dere)]
  |            ^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0635`.
error: could not compile `playground`

To learn more, run the command again with --verbose.
```

I think it would be much more helpful if the compiler complained about the unknown feature before it complains about me needing to add the feature. This led to a very confusing moment when I was missing the `s` at the end of `custom_test_frameworks` which the compiler didn't tell me about. That, or maybe it should just display both of those errors at once?
What I'm describing here is rather specific but maybe this can be made more general so that the compiler always rants about unknown attributes first? Not sure how feasible that is or how much sense that makes in other cases.

**Comment by eggyal** [comment]
> maybe this can be made more general so that the compiler always rants about unknown attributes first

Unknown *attributes* [already](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5e5494df2f837854ae80d2045d0d2887) trigger `cannot find attribute` errors.  The problem here is that the attribute `feature` is known, but *it* didn't complain about an unrecognised token in amongst its arguments.

**Comment by r00ster91** [comment]
Oh, right, I meant unknown attribute arguments, not unknown attribute.

**Comment by eggyal** [comment]
That's down to the implementation of each individual attribute, though... they receive an opaque token stream and can do whatever they want (or don't want) with it.

**Comment by eggyal** [comment]
Also, libraries can define their own `#![feature]`s so I don't think the compiler can in general know whether a given feature name exists or not (at least not without distinguishing builtin and library features in some way, which is not a bad idea but would be a breaking change).

**Comment by eggyal** [comment]
https://github.com/rust-lang/rust/blob/69e1d22ddbc67b25141a735a22a8895a678b32ca/compiler/rustc_passes/src/stability.rs#L957-L958

**Comment by eggyal** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/44232
